music_assistant/server/providers/opensubsonic/sonic_provider.py

```diff
@@ -97,6 +98,16 @@ async def handle_async_init(self) -> None:
             )
             raise LoginFailed(msg) from e
         self._enable_podcasts = self.config.get_value(CONF_ENABLE_PODCASTS)
+        try:
+            ret = self._conn.getOpenSubsonicExtensions()
```

Comment:
sorry, this is blocking IO again, wrap it in an executor task

Reply:
Please excuse my Python async I/O ignorance, but even if we make this call run in a different thread/process, this thread will block on the result but it shouldn't hold the GIL while blocking. Does the async I/O library run its own thread scheduler and the await keyword is conditional reschedule or yield by this thread in a way that executing native library code or a system call is not?

Reply:
You may simply not run ANY blocking IO code in an eventloop. You are calling (blocking!) code from an non-async library, in those cases you need to wrap it into an executor. I suggest you read a bit into asyncio and its event loop.

Check for example this video: https://www.youtube.com/watch?v=M-UcUs7IMIM